http/src/request/guild/create_guild.rs

```diff
@@ -111,20 +213,49 @@ impl<'a> CreateGuild<'a> {
 
     /// Set the channels to create with the guild.
     ///
-    /// The maximum number of channels that can be provided is 500.
+    /// The maximum number of channels that can be provided is 500. A `Vec` of
+    /// [`GuildChannelFragment`]s must be passed. The channels will be created in the order they
+    /// are passed. If inserting [`CategoryChannelFragment`]s, all child channels must be after the
+    /// category, and any non-children [`GuildChannelFragment`]s can not be between the children.
+    /// To specify `PermissionOverwrite`s, set the `PermissionOverwriteType` to `Role`, and its
+    /// `RoleId` to the id of a role you create with [`CreateGuild#roles`]. This `Vec` will be
+    /// transformed into temporary instances of each channel type, and sent to Discord for
+    /// validation.
     ///
     /// # Errors
     ///
     /// Returns [`CreateGuildError::TooManyChannels`] if the number of channels is over 500.
     ///
+    /// [`GuildChannelFragment`]: enum.GuildChannelFragment.html
+    /// [`CategoryChannelFragment`]: struct.CategoryChannelFragment.html
     /// [`CreateGuildError::TooManyChannels`]: enum.CreateGuildError.html#variant.TooManyChannels
-    pub fn channels(mut self, channels: Vec<GuildChannel>) -> Result<Self, CreateGuildError> {
+    /// [`CreateGuild#roles`]: struct.CreateGuild.html#method.roles
+    pub fn channels(
+        mut self,
+        fragments: Vec<GuildChannelFragment>,
+    ) -> Result<Self, CreateGuildError> {
         // Error 30013
         // <https://discordapp.com/developers/docs/topics/opcodes-and-status-codes#json>
-        if channels.len() > 500 {
-            return Err(CreateGuildError::TooManyChannels { channels });
+        if fragments.len() > 500 {
+            return Err(CreateGuildError::TooManyChannels {
+                channels: fragments,
+            });
         }
 
+        let mut channels: Vec<GuildChannel> = Vec::new();
+
+        for fragment in fragments.into_iter() {
+            let channel: GuildChannel = match fragment {
+                GuildChannelFragment::Category(c) => GuildChannel::Category(c.into()),
+                GuildChannelFragment::Text(t) => GuildChannel::Text(t.into()),
+                GuildChannelFragment::Voice(v) => GuildChannel::Voice(v.into()),
+            };
+
+            channels.push(channel);
+        }
+
+        println!("{:#?}", channels);
```

Comment:
Looks like this slipped in from debugging?

Reply:
It sure did.